src/Mod/Sketcher/Gui/DrawSketchHandlerBSplineByInterpolation.h

```diff
@@ -0,0 +1,606 @@
+                Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_mults = []");
+                Gui::Command::runCommand(Gui::Command::Gui, "_bsps = []");
+                for (auto & controlpoints: controlpointses) {
+                    // FIXME: can we get by without naming the variable?
+                    // TODO: variable degrees?
+                    QString cmdstr = QString::fromLatin1("_bsps.append(Part.BSplineCurve())\n"
+                                                         "_bsps[-1].interpolate(%1, PeriodicFlag=%2)\n"
+                                                         "_bsps[-1].increaseDegree(%3)")
+                        .arg(QString::fromLatin1(controlpoints.c_str()))
+                        .arg(QString::fromLatin1(ConstrMethod == 0 ?"False":"True"))
+                        .arg(myDegree);
+                    Gui::Command::runCommand(Gui::Command::Gui, cmdstr.toLatin1());
+                    // TODO: Adjust internal knots here (raise multiplicity)
+                    // How this contributes to the final B-spline
+                    if (controlpoints == controlpointses.front()) {
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_poles.extend(_bsps[-1].getPoles())");
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_knots.extend(_bsps[-1].getKnots())");
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_mults.extend(_bsps[-1].getMultiplicities())");
+                    }
+                    else {
+                        // TODO: Adjust for periodic
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_poles.extend(_bsps[-1].getPoles()[1:])");
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_knots.extend([_finalbsp_knots[-1] + i for i in _bsps[-1].getKnots()[1:]])");
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_mults[-1] = 3"); // FIXME: Hardcoded
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_mults.extend(_bsps[-1].getMultiplicities()[1:])");
+                    }
+                }
+
+                // TODO: Join the pieces of the B-splines
+                // {"poles", "mults", "knots", "periodic", "degree", "weights", "CheckRational", NULL};
+                Gui::cmdAppObjectArgs(sketchgui->getObject(), "addGeometry(Part.BSplineCurve"
+                                      "(_finalbsp_poles,_finalbsp_mults,_finalbsp_knots,%s,%d,None,False),%s)",
+                                      ConstrMethod == 0 ?"False":"True",
+                                      myDegree,
+                                      geometryCreationMode==Construction?"True":"False");
+                currentgeoid++;
+
+                // TODO: Confirm we do not need to delete individual elements
+                Gui::Command::runCommand(Gui::Command::Gui, "del(_bsps)\n");

```

Comment:
@abdullahtahiriyo I think I knot the answer already, but can you confirm if deleting just `_bsps` deletes each element of `_bsps`?

Reply:
My understanding is that del will decrease the reference counter of each element. In this case there is no other variable name referencing those element, then python will delete the elements.

So it would be a problem if you had:

```
mycoolbspline = Part.BSplineCurve...
mycoolarray = [mycoolbspline]
del(mycoolarray)
```

In this case, if you do not delete mycoolbspline that object remains in memory.

But do not hold me for a Python expert. I may be mistaken.

Reply:
Thought so too: this is as far as my knowledge stands. Does any of this change for our use case where there is some C++ involved as well? I'd imagine not, but you never know. Anyone we can ask?

Reply:
The c++ here is just calling the python. The objects are created in python as if you typed the code in the console. The fact that c++ code is writing to the python console is irrelevant.

Where it may be an issue is when you are writing the python bindings for a c++ object, then you need to decide whether the Python object takes a copy of the c++ object or not and then also decide on the ownership (we got leaks in the past there as it is indeed difficult to follow). But that is totally different scenario.

If you are unconvinced, feel free to post in the developers corner.

Reply:
OK. I think we should be fine so long as the C++ side itself doesn't create dangling objects.